### PR TITLE
Smartsheet IT Admin

### PR DESCRIPTION
**Pull request by sonaaru**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
N/A. New Integration.

## Description
Initial commit.

## Screenshots
N/A

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**
PANW IT team

### CODE REVIEW

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.yml [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      description: Gives the user information if the API is success else error information
+      type: String
+  dockerimage: demisto/smartsheet:1.0.0.9945
+  runonce: false
+  script: ''
+  subtype: python3
+  type: python
+fromversion: 5.0.0
+tests:
+- No tests (auto formatted)
```
Comment (dantavori) [inline Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.yml:247]: are we going to have an instance for this integration?

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+
+        return smartsheet_user
+
+
+class OutputContext:
+    """
+        Class to build a generic output and context.
+    """
+
+    def __init__(self, success=None, active=None, iden=None, username=None, email=None, errorCode=None,
```
Comment (dantavori) [inline Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py:147]: I think most of the attributes defined here are not really used anywhere, aren't they?

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+        generic_iam_context = OutputContext(success=False, iden=user_id, errorCode=errorCode,
+                                            errorMessage=errorMessage, details=res_json)
+
+    generic_iam_context_dt = "{}(val.id == obj.id && val.instanceName == obj.instanceName)".format(generic_iam_context.command)
+
+    outputs = {
+        generic_iam_context_dt: generic_iam_context.data
+    }
+
+    readable_output = tableToMarkdown('Get Smartsheet User:', generic_iam_context.data)
```
Comment (dantavori) [inline Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py:305]: maybe a better title would be something like "Smartsheet user {id} data". you might also use the tableToMarkdown attributes we talked about (headers, removeNull)

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+        generic_iam_context = OutputContext(success=False, iden=user_id,
+                                            errorCode=res_json.get('result').get('statusCode'),
+                                            errorMessage=errorMessage, details=res_json)
+
+    generic_iam_context_dt = "{}(val.id == obj.id && val.instanceName == obj.instanceName)".format(generic_iam_context.command)
+    outputs = {
+        generic_iam_context_dt: generic_iam_context.data
+    }
+
+    readable_output = tableToMarkdown('Remove Smartsheet User:', generic_iam_context.data)
```
Comment (dantavori) [inline Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py:351]: same here and in the following commands

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+## [Unreleased]
```
[the remark this answers is not in the document]
Comment (sonaaru) [reply Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/CHANGELOG.md]: Removed

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/README.md [hunk 6]
```diff
@@ -0,0 +1,387 @@
+1.	Get / Retrieve user information from Smartsheet
+2.	Create a user in Smartsheet
+3.	Update a user
+4.	Disable a user
+This integration was integrated and tested with version xx of Smartsheet IT Admin
```
[the remark this answers is not in the document]
Comment (sonaaru) [reply Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/README.md]: Updated

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.yml (hunk 1 above)
Comment (sonaaru) [reply Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.yml:247]: We have an instance in the dev environment but it was for free trial. We will create a new one going forward.

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py (hunk 2 above)
Comment (sonaaru) [reply Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py:147]: All the attributes here are being use somewhere in the code. You can refer to line 288 and line 291.

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py (hunk 3 above)
Comment (sonaaru) [reply Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py]: Updated

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py (hunk 4 above)
Comment (sonaaru) [reply Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.py]: Updated

File: Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.yml [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+    - contextPath: EnableUser.errorCode
+      description: 'HTTP error response code '
+      type: Number
+    - contextPath: EnableUser.errorMessage
+      description: Reason why the API is failed
+      type: String
+    - contextPath: EnableUser.details
+      description: Gives the user information if the API is success else error information
+      type: String
+  dockerimage: demisto/smartsheet:1.0.0.9945
```
Comment (dantavori) [inline Packs/SmartSheetITAdmin/Integrations/Smartsheet_IT_Admin/Smartsheet_IT_Admin.yml:240]:
```suggestion
  dockerimage: demisto/smartsheet:1.0.0.10255
```

**Pull request merged**